distributed/scheduler.py

```diff
@@ -904,13 +904,14 @@ def remove_worker(self, comm=None, address=None, safe=False, close=True):
             del self.worker_comms[address]
             del self.ncores[address]
             self.workers.remove(address)
-            del self.aliases[self.worker_info[address]['name']]
+            _ = self.coerce_address(self.worker_info[address]['name'], clear_cache=True)
```

Comment:
It seems odd to me to call coerce address in order to invalidate a cache.  I would also expect any call to coerce address to add the new value back into the cache after invalidation.  

In either case do we still delete the name from self.aliases?

Reply:
The caching happens after the logic to strip off the protocol and port – the choice was either to have it in coerce_address, which already manages this stripping, or to duplicate this logic in a separate function. (Or to factor out this logic and use it in both).